src/peer_manager.rs

```diff
@@ -661,15 +661,31 @@ impl PeerManager {
             match peer.state {
                 PeerState::ConnectionInfoPreparing { .. } |
                 PeerState::ConnectionInfoReady(_) |
-                PeerState::CrustConnecting |
                 PeerState::SearchingForTunnel |
-                PeerState::Routing(_) => {
-                    trace!("{:?} Unexpected peer state {:?} while adding {:?} to routing table.",
+                PeerState::Routing(RoutingConnection::Proxy(_)) |
+                PeerState::Routing(RoutingConnection::Direct) |
+                PeerState::Routing(RoutingConnection::Tunnel) => {
+                    trace!("{:?} Unexpected peer state {:?} while adding {:?} into routing table.",
+                           self,
+                           peer.state,
+                           peer_id)
+                }
+                PeerState::CrustConnecting if !self.unknown_peers.contains_key(peer_id) => {
```

Comment:
I thought this too, but I think the `if` guard makes it trouble-some (it would apply to *all* the patterns that are joined by `|`).